src/haven/devices/ion_chamber.py

```diff
@@ -270,6 +271,36 @@ async def connect(
                 # Update the labjack's input's .DESC field to match the scaler channel
                 await self.voltmeter_channel.description.set(desc)
 
+    async def default_timeout(self) -> float:
+        """Calculate the expected timeout for triggering this ion chamber.
+
+        If ``self.mcs.scaler.preset_time`` is set and the first
+        (clock) channel is serving as a gate
+        (``self.mcs.scaler.channels[0].is_gate.get_value() == True``)
+        then the timeout is a little longer than the preset
+        time. Otherwise the timeout is calculated based on the longest
+        time the scaler can count for based on when the channel 0
+        clock register would fill up.
+
+        Returns
+        =======
+        timeout
+          The optimal timeout for trigger this ion chamber's scaler.
+
+        """
+        aws = [
+            self.mcs.scaler.channels[0].is_gate.get_value(),
+            self.mcs.scaler.preset_time.get_value(),
+            self.mcs.scaler.clock_frequency.get_value(),
+        ]
+        is_time_limited, count_time, clock_freq = await asyncio.gather(*aws)
+        if is_time_limited:
+            # We're using the preset time to decide when to stop
+            return count_time + DEFAULT_TIMEOUT
+        else:
+            max_count = 2**self._clock_register_width / clock_freq
+            return max_count + DEFAULT_TIMEOUT
```

Comment:
Good question.

> What is our clock_frequency?

It's the value circled in red in the image below. There's a dedicated clock that the scaler uses to know how much time has passed. In the below example, the clock is set to ``9.6e6`` Hz. When we ask the scaler to count to 300 seconds, then it uses this clock and will count for ``300 * 9.6e6`` = ``2880000000`` clock cycles, by making this the preset value for channel 1 on the scaler.

![image](https://github.com/user-attachments/assets/eb04d1f3-897a-483e-aec2-4627eb0e17fd)

> Is it a constant value depending on the scaler hardware? Or a dynamic value?

Whether it's changeable depends on where it comes from. The Struck 3820 VME scaler has an internal clock that is always 50 MHz. The scaler's internal registers are 32 bits wide, so the biggest number that it can keep track of is ``2**32`` == ``4294967296``. This means that if the clock is 50 MHz, the longest the scaler can count for is ``2**32 / 5e7`` = ``85.9 seconds``. This is too short for us and since we can't change the internal registers then we need a slower clock. 

> When is the value changed?

We're moving towards the MCS counter to get away from VME, and these scalers don't have an internal clock. Instead they have 4 frequency generators whose frequency can be set in EPICS, and so use one of these as an external clock. We can set it to as low a frequency as needed to count for the length of time we need.

It's probably not changed very often, but it can be changed when needed for long scans. Eventually I envision Haven's IonChamber device being able to set the clock frequency dynamically based on the preset time requested for the measurement.